File: LeetCode/33. Search in Rotated Sorted Array.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        left = 0 
        right = len(nums) - 1
        while left <= right:
            if nums[left] == target:
                return left 
            if nums[right] == target:
                return right 
            left += 1 
            right -= 1
        return -1 
x = Solution()
print(x.search([4,5,6,7,0,1,2],0))
print(x.search([4,5,6,7,0,1,2],3))

File: LeetCode/33. Search in Rotated Sorted Array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    for(let i = 0 ; i < nums.length; i++) {
        if(nums[i] == target){
            return i 
        }
    }
    return - 1
}

console.log(search([4,5,6,7,0,1,2], 0)) // 4
console.log(search([4,5,6,7,0,1,2], 3)) // -1
console.log(search([1], 0)) // -1   
console.log(search([1], 1)) // 0